python/cudf/cudf/dataframe/columnops.py

```diff
@@ -553,6 +556,25 @@ def as_column(arbitrary, nan_as_null=True, dtype=None, name=None):
     return data
 
 
+def as_nvstrings(arbitrary):
+    representative = None
+    for x in arbitrary:
+        if x is not None:
+            representative = x
+            break
```

Comment:
This is insanely expensive in the case of `[None, None, None, None,..., 'a']`, can we not just shepherd this through something else?

Reply:
Yeah I hate this too. Do we have GPU-accelerated type inference anywhere? Otherwise I'm not sure how we can match the Pandas casting behavior accepting arbitrary python lists as the search values to `isin()`:
https://github.com/rapidsai/cudf/blob/6686b158f69956bd81afef50f22ffc13a529df92/python/cudf/cudf/dataframe/series.py#L1804-L1816

Reply:
We should just do `columnops.as_column(input, dtype="object")` which should handle coercing to an object dtype.

Reply:
I was able to remove this code now that we can cast any type to any other via `astype()` :tada: